code/obj/artifacts/artifactprocs.dm

```diff
@@ -4,35 +4,19 @@
 	if (!istype(T,/turf/) && !istype(T,/obj/))
 		return
 
-	var/rarityroll = 1
-
-// artifact tweak. rarity 1 now contains garbage artifacts so that it's easier to control how much garbage science sees.
-	switch(rand(1,100))
-		if (36 to 80) 		// 45%. 4% chance for a particular level 2 art.
-			rarityroll = 2
-		if (81 to 95) 		// 15%. With current art list this means 2% chance of a certain level 3 art
-			rarityroll = 3
-		if (96 to 100) 		// 5%. With current art list this means 1% chance of a certain level 4 art. 2 of the 5 are bombs...
-			rarityroll = 4
-		else 							// 35%. 4% chance for a particular garbage level 1 art.
-			rarityroll = 1
-
-	var/list/selection_pool = list()
+	var/list/artifactweights
+	if(forceartiorigin)
+		artifactweights = artifact_controls.artifact_rarities[forceartiorigin]
+	else
+		artifactweights = artifact_controls.artifact_rarities["all"]
 
+	var/datum/artifact/picked
 	if(forceartitype)
-		selection_pool += forceartitype
+		picked = forceartitype
 	else
-		for (var/datum/artifact/A as() in concrete_typesof(/datum/artifact))
-			if (initial(A.rarity_class) != rarityroll)
-				continue
-			if (istext(forceartiorigin) && !(forceartiorigin in initial(A.validtypes)))
-				continue
-			selection_pool += A
-
-	if (selection_pool.len < 1)
-		return
-
-	var/datum/artifact/picked = pick(selection_pool)
+		if (artifactweights.len < 1)
```

Comment:
I mean this works but still it's a bit of a weird way to check for an empty list. :shelterfrog: